### DISCUSSION TITLE
PropertyGrid has a bug on child form at version 6.0.0-beta8

### DISCUSSION THREAD

**Discussion by Dylan970811** [kikipoulet/SukiUI#324] [Q&A]
When I open a 'child form' from the 'father form' by 'PropertyGrid'，what it shows is not the 'child form' but  another 'father form'.  Whether it's a bug or the way code works is not right?
![PixPin_2024-11-07_14-14-06](https://github.com/user-attachments/assets/38680f93-4dcc-4990-989b-6e9bbaa0c08b)

**Comment by kikipoulet** [comment]
It's not a bug, it's a feature 😅

You have an object shown in the PropertyGrid, and this field is an object from a class of yours, it's not like showing a string or an int. As it is too complex to be shown in one line, you can click on the "more info" button to have a new window opening with its own PropertyGrid instance of the child object.

**Reply by Dylan970811** [reply]
I got how it works. In the question, I have difined a class(A) that conatining another class's(B) instance as its one of properties. The B class's instance will be shown as a button "More Info" in the father form. And I hope it will show the B class's own PropertyGrid  on the child form rather than still a A class's PropertyGrid when I click the button "More Info".  But what it shows is still class A's PropertyGrid on the child form. Or does it mean that the new child form can only show the class A's PropertyGrid again?

**Reply by Insire** [reply]
This is actually broken. It worked with the sukihost, but since thats gone now (not implemented anymore, but commented out) opening the child form in a new dialog window, will result in obviously wrong behavior. I already had look last night, but it was late and i couldnt quite wrap my head around anymore how this was supposed to work. I'll take a look again today.

**Reply by Insire** [reply]
I opened a PR #327 that adresses this.

**Reply by sirdoombox** [reply]
This one was probably on me, when I reimplemented the host stack I just "fixed" propertygrid, not actually knowing how it worked having not really spent any time with it.

Sorry!

**Reply by Insire** [reply]
> This one was probably on me, when I reimplemented the host stack I just "fixed" propertygrid, not actually knowing how it worked having not really spent any time with it.
> 
> Sorry!

No worries, reflection stuff rarely is straight forward.
